R.swift/Generators/StringsGenerator.swift

```diff
@@ -0,0 +1,313 @@
+//
+//  StringsGenerator.swift
+//  R.swift
+//
+//  Created by Nolan Warner on 2016/02/23.
+//  Copyright © 2016 Nolan Warner. All rights reserved.
+//
+
+import Foundation
+
+struct StringsGenerator: Generator {
+  let externalStruct: Struct?
+  let internalStruct: Struct? = nil
+
+  init(localizableStrings: [LocalizableStrings]) {
+
+    let localized = localizableStrings.groupBy { $0.filename }
+    let groupedLocalized = localized.groupBySwiftNames { $0.0 }
+
+    for (sanitizedName, duplicates) in groupedLocalized.duplicates {
+      warn("Skipping \(duplicates.count) strings files because symbol '\(sanitizedName)' would be generated for all of these filenames: \(duplicates.joinWithSeparator(", "))")
+    }
+
+    let empties = groupedLocalized.empties
+    if let empty = empties.first where empties.count == 1 {
+      warn("Skipping 1 strings file because no swift identifier can be generated for filename: \(empty)")
+    }
+    else if empties.count > 1 {
+      warn("Skipping \(empties.count) strings files because no swift identifier can be generated for all of these filenames: \(empties.joinWithSeparator(", "))")
+    }
+
+    externalStruct = Struct(
+      comments: ["This `R.string` struct is generated, and contains static references to \(groupedLocalized.uniques.count) localization tables."],
+      type: Type(module: .Host, name: "string"),
+      implements: [],
+      typealiasses: [],
+      properties: [],
+      functions: [],
+      structs: groupedLocalized.uniques.flatMap(StringsGenerator.stringStructFromLocalizableStrings)
+    )
+  }
+
+  private static func stringStructFromLocalizableStrings(filename: String, strings: [LocalizableStrings]) -> Struct? {
+
+    let name = sanitizedSwiftName(filename)
+    let params = computeParams(filename, strings: strings)
+
+    return Struct(
+      comments: ["This `R.string.\(name)` struct is generated, and contains static references to \(params.count) localization keys."],
+      type: Type(module: .Host, name: name),
+      implements: [],
+      typealiasses: [],
+      properties: params.map(StringsGenerator.stringLet),
+      functions: params.map(StringsGenerator.stringFunction),
+      structs: []
+    )
+  }
+
+  // Ahem, this code is a bit of a mess. It might need cleaning up... ;-)
```

Comment:
Sure, although most of the code in this method is related generating warnings. This should be cleanup as part of:
- [Add support for localised storyboards and nibs](https://github.com/mac-cain13/R.swift/issues/155)
- [Refactor warnings so that it's not an sideeffect anymore](https://github.com/mac-cain13/R.swift/issues/136)